cmd/kubelet/app/server.go

```diff
@@ -986,16 +1003,19 @@ func InitializeTLS(kf *options.KubeletFlags, kc *kubeletconfiginternal.KubeletCo
 
 // setContentTypeForClient sets the appropritae content type into the rest config
 // and handles defaulting AcceptContentTypes based on that input.
-func setContentTypeForClient(cfg *restclient.Config, contentType string) {
+func setContentTypeForClient(cfgs *restclient.Config, contentType string) {
 	if len(contentType) == 0 {
 		return
 	}
-	cfg.ContentType = contentType
-	switch contentType {
-	case runtime.ContentTypeProtobuf:
-		cfg.AcceptContentTypes = strings.Join([]string{runtime.ContentTypeProtobuf, runtime.ContentTypeJSON}, ",")
-	default:
-		// otherwise let the rest client perform defaulting
+
+	for _, cfg := range cfgs.GetAllConfigs() {
+		cfg.ContentType = contentType
+		switch contentType {
+		case runtime.ContentTypeProtobuf:
+			cfg.AcceptContentTypes = strings.Join([]string{runtime.ContentTypeProtobuf, runtime.ContentTypeJSON}, ",")
```

Comment:
The standard protobuf content type is "application/json".  There are limited type of accept content types for different components, such as "application/vnd.google.protobuf" for Prometheus. In that case, I am worried that there are duplicate entries in runtime.ContentTypeProtobuf, such as "application/json,application/json"